### DISCUSSION TITLE
Test Jamulus on MacOS VM with Jack server on dummy driver

### DISCUSSION THREAD

**Discussion by newlaurent62** [jamulussoftware/jamulus#1683] [Technical Support & Help]
Hello,

I installed MacOS Mojave in a virtual machine. This VM does not have audio device so i use Jack server with dummy driver.

When i start Jamulus app on this VM it complains there is no coreaudio device (and that's true !) and the application exits immediately. But a jack server is running on dummy driver and i have no chance to use Jamulus with the Jack server ...

Is there a way to force the use of the jack server with Jamulus on MacOS Mojave even if there is no coreaudio device ?

Note : It's the first time i use MacOS, so i have no experience with Jamulus on MacOS.

Kind Regards,
Laurent

**Comment by hoffie** [comment]
Hi,

sorry to be that guy, but essentially this has a really similar root cause to #1678: The official Mac builds use coreaudio. There is support for jack on Mac, but it has to be enabled during build time via `CONFIG+=jackonmac`. Currently, there are no official builds with this configuration. :(

I guess one workaround could be to find a virtual Coreaudio device for Mac?

Kind regards
Christian

**Reply by newlaurent62** [reply]
Thanks for the quick answer.

My group of musicians uses the three plateforms : Mac, Windows and Linux. My aim is to use Jack server on all three plateforms.

This would allow us to use Jamulus (for musicians) and QJackTirip (for sound engineer) at the same time ... But it's seems more difficult than i imagined.

----

I could not compile a program on the MacOS VM. (The xcode from app store requires Big Sur and my VM is Mojave). May be i should install a big sur VM ... But i heard that Big Sur has problem with audio applications (Is that still true ?)

**Reply by henkdegroot** [reply]
You can still download/install xcode 11.x you don't need to have Big Sur. You just need to go to the "more" section to download:
[https://developer.apple.com/download/more/](https://developer.apple.com/download/more/)

**Reply by newlaurent62** [reply]
Thanks @henkdegroot for the tips.

**Reply by henkdegroot** [reply]
@newlaurent62 just out of interest, what platform do you use as the host for the virtual machine and what virtual machine software?

**Reply by newlaurent62** [reply]
@henkdegroot  https://github.com/myspaghetti/macos-virtualbox/releases on Linux debian.

**Reply by newlaurent62** [reply]
Hello,

Is there a tutorial on how to install Jamulus from source on MacOS Mojave ?

I compiled Jamulus for jack on MacOS and can run it without problem from the terminal ...

How can i have an icon for Jamulus in Applications folder ?

Can i create easily create a dmg usable on all my group mac computers ?

**Reply by hoffie** [reply]
It should be possible to create builds similar to the official ones by running the relevant scripts (maybe slightly modified for your use case). The relevant job is here and points to the other related scripts:

https://github.com/jamulussoftware/jamulus/blob/a208a11182929d2c3259376502aa86af1742842a/.github/workflows/autobuild.yml#L122

This should also include the icon in the Applications folder (if this works in the official release, which I assume).

Does that help?

**Reply by newlaurent62** [reply]
Yes, i created a dmg file and installed Jamulus from it but now i have to modify the mac/deploy_mac.sh file with CONFIG+=jackonmac instead on CONFIG+=release and i hope this will do the trick ...

**Reply by newlaurent62** [reply]
Ok now i have a dmg install file for jackonmac Jamulus app. It works.

**Reply by newlaurent62** [reply]
https://github.com/jamulussoftware/jamulus/discussions/1712
